sycl/test-e2e/USM/P2P/p2p_atomics.cpp

```diff
@@ -50,15 +50,15 @@ int main() {
     h_sum += value;
   }
 
-  int *d_sum = malloc_shared<int>(1, Queues[0]);
+  int *d_sum = malloc_device<int>(1, Queues[0]);
```

Comment:
Should we first check if `malloc_device` is even supported on the device? Like:
```
if (dev.get_info<info::device::usm_device_allocations>())
...
```

Reply:
That would be a separate change - we're already using it on the next line.

Reply:
@JackAKirk Please update this test (and other P2P tests, if applicable) to also check if device allocations are supported on the device(s).